### PR TITLE
GH-33783: [C#] Update release verification to use .NET 7.0

### LINKED ISSUE #33783: [Release][C#] Release verification tasks fail with new version of C# 7.0.x
### Describe the bug, including details regarding any error messages, version, and platform.

dotnet 7.0.0 was released on November and we never added new tasks for it. https://dotnet.microsoft.com/en-us/download/dotnet/7.0
We currently only have verification jobs with 6.0.202. If I run verification locally with Ubuntu 22.04 with .NET 6.0.202 C# jobs are successful:
```
===========================
Build and test C# libraries
===========================
└ Ensuring that C# is installed...
└ Installed C# at  (.NET 6.0.202)
You can invoke the tool using the following command: sourcelink
Tool 'sourcelink' (version '3.1.1') was successfully installed.
/tmp/arrow-11.0.0.ReaWC/apache-arrow-11.0.0/csharp /tmp/arrow-11.0.0.ReaWC/apache-arrow-11.0.0 ~/code/arrow
  Determining projects to restore...
```

but it fails if I upgrade dotnet to `7.0.102`:
```
===========================
Build and test C# libraries
===========================
└ Ensuring that C# is installed...
└ Found C# at  (.NET 7.0.102)

Welcome to .NET 7.0!
---------------------
SDK Version: 7.0.102
...
dev/release/verify-release-candidate.sh: line 341: 129149 Segmentation fault      (core dumped) dotnet tool install --tool-path ${csharp_bin} sourcelink
Failed to verify release candidate. See /tmp/arrow-11.0.0.lNQyX for details.
```

### Component(s)

C#, Release

### PR DESCRIPTION
**Pull request by westonpace**
Update verification script to download and install 7.0 but run tests against 6.0 if the 6.0 SDK is present.  Update examples to use 7.0 only.
* Closes: #33783

### CODE REVIEW

**Comment by westonpace** [comment]
@github-actions crossbow submit --group verify-rc-source --param release=11.0.0 --param rc=0

**Comment by westonpace** [comment]
@github-actions crossbow submit --group verify-rc-binaries --param release=11.0.0 --param rc=0

**Comment by westonpace** [comment]
@github-actions crossbow submit --group verify-rc-binaries

**Comment by westonpace** [comment]
@github-actions crossbow submit --group verify-rc-source

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -347,18 +347,15 @@ install_csharp() {
 
   show_info "Ensuring that C# is installed..."
 
-  if which dotnet > /dev/null 2>&1; then
+  if dotnet --version | grep 7\.0 > /dev/null 2>&1; then
```
Comment (eerhardt) [inline dev/release/verify-release-candidate.sh:350]: what happens if `dotnet` is not installed? Does this script error out?

**Review by eerhardt — COMMENTED** [review]
I almost wonder if we need both 6.0 and 7.0.

Would it be better to just require 7.0 when doing development on the library?

**Comment by westonpace** [comment]
> I almost wonder if we need both 6.0 and 7.0.

Are you asking if it is worth testing with 6.0?  I agree that may be excessive.  I was erring on the side of caution but if you want me to refine this so its a strict upgrade to 7.0 I'd be fine with that.

> Would it be better to just require 7.0 when doing development on the library?

You should be able to get away with doing development with just 7.0.  The only downside will be that you will need to make sure to run your tests with `dotnet test --framework net7.0` or else it will try and run them for each target and fail on 6.0 (I believe Visual Studio IDE makes this a drop down and only tests a single framework at a time).

File: dev/release/verify-release-candidate.sh (hunk 1 above)
Comment (westonpace) [reply dev/release/verify-release-candidate.sh:350]: The `if` fails and it downloads and installs a temporary local copy of .NET 7.

**Comment by eerhardt** [comment]
By default, VS will run both:

image

**Comment by westonpace** [comment]
@eerhardt circling back on this.  Would you rather we just upgrade everything to .net7 (and remove references to .net6) to keep the developer experience simple?

**Comment by eerhardt** [comment]
I'm fine with having both .NET 6 and .NET 7 support, I guess, if we need it.

Maybe let's just do .NET 7 for now. And if someone finds a reason to keep supporting .NET 6, we can add it then?

**Comment by westonpace** [comment]
@github-actions crossbow submit --group verify-rc-source-csharp

**Comment by westonpace** [comment]
@github-actions crossbow submit --group verify-rc-source

**Comment by westonpace** [comment]
> Maybe let's just do .NET 7 for now. And if someone finds a reason to keep supporting .NET 6, we can add it then?

Done.  CI failures are unrelated and the wheel verification builds seem to be passing.  I'm going to go ahead and merge this.

**Pull request merged**